uefi/src/proto/boot_policy.rs

```diff
@@ -0,0 +1,108 @@
+//! Module for the [`BootPolicy`] helper type.
+
+use core::fmt::{Display, Formatter};
+
+/// Errors that can happen when working with [`BootPolicy`].
+#[derive(Debug, Copy, Clone, PartialOrd, PartialEq, Eq, Ord)]
+pub enum BootPolicyError {
+    /// Only `0` and `1` are valid integers, all other values are undefined.
+    InvalidInteger(u8),
+}
+
+impl Display for BootPolicyError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        let s = match self {
+            Self::InvalidInteger(_) => {
+                "Only `0` and `1` are valid integers, all other values are undefined."
+            }
+        };
+        f.write_str(s)
+    }
+}
+
+#[cfg(feature = "unstable")]
+impl core::error::Error for BootPolicyError {}
+
+/// The UEFI boot policy is a property that influences the behaviour of
+/// various UEFI functions that load files (typically UEFI images).
+///
+/// This type is not ABI compatible. On the ABI level, this is an UEFI
+/// boolean.
```

Comment:
It seems like it could be made compatible: add `repr(u8)` and `BootSelection = 1`, `ExactMatch = 0`.

(As long as this type is only passed into UEFI APIs, and not read from them; in that case `newtype_enum` might be better.)

Reply:
Do we want to postpone that decission/discussion to #1307?

Reply:
As long as the type is only being passed into the API, UB shouldn't be a concern -- we'll always initialize `BootPolicy` correctly.

Reply:
Were you just adding thoughts to the discussion or do you want this comment to be removed before we merge it? That's not clear to me right now :)

Reply:
I was thinking we should make changes before merging, but actually since there is no declared `repr` right now, it wouldn't be a breaking change to add `repr(u8)` in the future I think. So let's go ahead and merge.